### ISSUE TITLE
code: 'METHOD.NOT-SUPPORTED'

### ISSUE THREAD

**Issue by grzegorz914** [fdegier/homebridge-jablotron-alarm#58]
@F4stFr3ddy 
from time to time this error appears in the log

```
[24.08.2021, 10:52:41] [Alarm] {
  'http-code': 200,
  errors: [
    {
      code: 'METHOD.NOT-SUPPORTED',
      message: 'Unknown method.',
      debug: 'Unknown method.'
    }
  ]
}
[24.08.2021, 10:52:41] [Alarm] Error: 
    at /usr/lib/node_modules/homebridge-jablotron/lib/jablotron-client.js:25:31
    at IncomingMessage.<anonymous> (/usr/lib/node_modules/homebridge-jablotron/lib/jablotron-client.js:57:21)
    at IncomingMessage.emit (events.js:412:35)
    at endReadableNT (internal/streams/readable.js:1317:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
```

**Comment by F4stFr3ddy** [comment]
Might have to add a bit more debugging to find where this is coming from. I have never seen it in my logs nor experienced an issue. When this occurs, do you experience any issues?

**Comment by grzegorz914** [comment]
> I see the same error, almost daily, and If it occurs it is at the same time:
> 
> [29/10/2021, 04:40:24] [Jablotron] { 'http-code': 200, errors: [ { code: 'METHOD.NOT-SUPPORTED', message: 'Unknown method.', debug: 'Unknown method.' } ] } [29/10/2021, 04:40:24] [Jablotron] Error: at /usr/lib/node_modules/homebridge-jablotron/lib/jablotron-client.js:25:31 at IncomingMessage. (/usr/lib/node_modules/homebridge-jablotron/lib/jablotron-client.js:57:21) at IncomingMessage.emit (events.js:412:35) at endReadableNT (internal/streams/readable.js:1317:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)

Exactly the error appear on my side in the same time, that must be something wrong with Jablotron server side.

**Comment by fdegier** [comment]
*Situation*
I have given it some thought and come to the following. The problem seems to be on Jablotron side, for some reason at this time, and perhaps some other random times, the request send to Jablotron results in unexpected behavior, due to the error handling already in place, the error is caught and normal operation continues. This suspicion is strenghtened by the fact that we receive a 200.

*Possible solutions*
- Catch the error explicitly, this still gives an error but it acknowledges in the code that we know this is an error and we accept it. 
- Catch the error and retry the request, this will result in a slightly longer operation and might still lead to the same error, not sure how long Jablotron has some issue. 


*Questions*
- Do you guys have alerting on errors? Personally I do not have that so I am never notified of this being an issue, besides that I will be sleeping at that time.

*Source code*
```javascript
this.doRequest(endpoint, payload, cookies, function (response) {
            if (response['status']) {
                successCallback(response);
            } else if (response['error_message'] == 'Operation failed') {
                errorCallback(new Error('Operation failed'))
            } else {
                self.log(response);
                errorCallback(new Error(response['error_status']))
            }
```


What do you guys think? Am I overlooking a possibility here?

**Comment by grzegorz914** [comment]
No, I don't have alerting notification, for me this error can be omitted if happens.
This error have no impact on system functionality at this time.

**Comment by fdegier** [comment]
@grzegorz914  can you check #59 ?

**Comment by grzegorz914** [comment]
Look’s good to me

**Comment by grzegorz914** [comment]
@fdegier 

when You push fixed version 2.3.1 to NPM?

**Comment by fdegier** [comment]
I wanted to test it on my installation first but haven't gotten around to doing that. I will test it today and release.
